packages/next-auth/src/nextAuthPlugin.ts

```diff
@@ -3,39 +3,71 @@ import {
   PartialStlContext,
   MakeStainlessPlugin,
   Params,
+  StlContext,
 } from "stainless";
 import { IncomingMessage, ServerResponse } from "http";
 import { NextServerContext } from "@stl-api/next";
 import { AuthOptions, Session, getServerSession } from "next-auth";
 
+/// An authenticated session, with user data available.
+export type UserSession = Session & {
+  user: {
+    name?: string | null
+    email?: string | null
+    image?: string | null
+  }
+}
+
 declare module "stainless" {
   interface StlContext<EC extends AnyEndpoint> {
-    session?: Session | null;
+    /// If a route is authenticated, handlers will always have access to a valid user session.
+    /// Otherwise, a session is provided if a user is logged in.
+    session: EC['config'] extends { authenticated: true } ? UserSession : UserSession | undefined;
```

Comment:
Should this be
```suggestion
    session: EC['config'] extends { authenticated: true } ? UserSession : (Session | null);
```
?

Reply:
For null see typing comment elsewhere, right now the logic never sets `session` if it wouldn't have a user. This goes back to another comment: what behavior do we want around users? I'll get more info from NextAuth docs, then we should talk more in person.

Reply:
ah, I would guess we would want to set `session` if it exists regardless (and perhaps `null` otherwise, to indicate explicitly there is none)